### ISSUE TITLE
Document and port kFreeBSD backend 

### ISSUE THREAD

**Issue by avsm** [mirage/mirage#186]
Notes from @pgj on the kFreeBSD backend, and how to get it working using Mirage 0.9.4.  This needs porting to the new world order in 1.0, but the existing instructions are a great start:

```
export ASSUME_ALWAYS_YES=1
export PACKAGESITE=http://csupor.inf.elte.hu/mirage/freebsd-pkg/latest/
export OPAMYES=1
CCOMP=4.01.0beta1+mirage-kfreebsd

pkg install ocaml-opam gmake git rsync
# must be also present: kernel sources
opam init
opam repo add mirage     git://github.com/mirage/opam-repo-dev
opam repo add pgj-mirage git://github.com/pgj/opam-repo-dev
opam repo add pgj-opam   git://github.com/pgj/opam-repo
opam switch ${CCOMP}
eval `opam config env`
opam install oasis
opam install mirage-kfreebsd
opam install mirage-www
kldload /root/.opam/${CCOMP}/bin/mir-www.ko

# networking
HOSTIF=xn0
MIRGW=10.0.0.1
MIRIP=10.0.0.2
NETMASK=255.255.255.0
NETWORK=10.0.0.0/24

ifconfig ${HOSTIF} alias ${MIRGW} netmask ${NETMASK}
sysctl net.inet.ip.forwarding=1
kldload pf
pfctl -f- << EOF
nat on ${HOSTIF} from { ${NETWORK} } to any -> (${HOSTIF})
rdr on ${HOSTIF} inet proto tcp to port 80 -> ${MIRIP} port 80
EOF
pfctl -e
```

Remarks from Gabor:
- I have prebuilt the FreeBSD packages using pkg(8), the next generation
  package manager and made them available on my department machine.  It takes
  some time to download the packages through Rackspace, but it is still quicker
  than building everything from sources.  I had to do that because the FreeBSD
  Project officially does not provide prebuilt pkg(8) packages (yet) -- once
  they are any, the PACKAGESITE variable can be adjusted.  (ASSUME_ALWAY_YES
  sets the non-interactive operation for pkg(8).)
- As I wrote in the comment, it is expected that the operating system is
  installed together with the kernel sources.
- I also my own OPAM package repositories for the new package,
  mirage-kfreebsd, and for all the modifications I have made to the upstreams.
  They might be able go away once everything is merged into upstream or new
  versions for the patched packages are published.
- I had to install oasis because the new version of optcomp will not be able
  to configure without that.
- Regarding mirage-www, I decided to go for the "transparent patching"
  approach, that is, I am pulling the updates from the mirage/mirage-www
  repository, then I apply the kFreeBSD patches on top of that.  So it shall
  always get the latest contents from the repository, without requiring me to
  transfer the commits from upstream to my clone of mirage-www.  However, if
  the patches cannot be applied any more, it will fail to build.  I guess,
  this will not be an issue after the kFreeBSD backend is merged.
- I use PF available on the host to do the network plumbing required for the
  Mirage instance, so I included the very simple PF NAT rules in the script.

**Comment by yomimono** [comment]
I think this is no longer relevant now that we have solo5+bhyve; please reopen and comment if you disagree.
